cluster-agent/controllers/argoproj.io/namespace_reconciler_test.go

```diff
@@ -951,4 +951,188 @@ var _ = Describe("Namespace Reconciler Tests.", func() {
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      argosharedutil.GenerateArgoCDRepoCredSecretName(repositoryCredentials),
+					Namespace: gitopsEngineInstance.Namespace_name,
+				},
+			}
+		})
+
+		It("Should not create Operation for Secret, even if it is missing in Cluster, since RepositoryCredentials is created recently.", func() {
+
+			defer dbq.CloseDatabase()
+
+			By("Get list of Operations before calling function.")
+
+			var operation []db.Operation
+
+			err := dbq.ListOperationsByResourceIdAndTypeAndOwnerId(ctx, repositoryCredentials.RepositoryCredentialsID, db.OperationResourceType_RepositoryCredentials, &operation, db.SpecialClusterUserName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(operation).To(BeEmpty())
+
+			By("Call function to recreate Secret if missing from cluster.")
+
+			recreateClusterSecrets_RepositoryCredentials(ctx, dbq, k8sClient, log)
+
+			By("Get list of Operations after calling function.")
+
+			err = dbq.ListOperationsByResourceIdAndTypeAndOwnerId(ctx, repositoryCredentials.RepositoryCredentialsID, db.OperationResourceType_RepositoryCredentials, &operation, db.SpecialClusterUserName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(operation).To(BeEmpty())
+		})
+
+		It("Should not create Operation for Secret, since it is already present in Cluster.", func() {
+
+			defer dbq.CloseDatabase()
+
+			By("Set 'Created_on' field more than 30 Minutes.")
+
+			repositoryCredentials.Created_on = time.Now().Add(time.Duration(-(31 * time.Minute)))
+			err := dbq.UpdateRepositoryCredentials(ctx, &repositoryCredentials)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Create Secret CR in cluster.")
+
+			err = k8sClient.Create(ctx, &secret)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Get list of Operations before calling function.")
+
+			var operation []db.Operation
+
+			err = dbq.ListOperationsByResourceIdAndTypeAndOwnerId(ctx, repositoryCredentials.RepositoryCredentialsID, db.OperationResourceType_RepositoryCredentials, &operation, db.SpecialClusterUserName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(operation).To(BeEmpty())
+
+			By("Call function to recreate Secret if missing from cluster.")
+
+			recreateClusterSecrets_RepositoryCredentials(ctx, dbq, k8sClient, log)
+
+			By("Get list of Operations after calling function.")
+
+			err = dbq.ListOperationsByResourceIdAndTypeAndOwnerId(ctx, repositoryCredentials.RepositoryCredentialsID, db.OperationResourceType_RepositoryCredentials, &operation, db.SpecialClusterUserName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(operation).To(BeEmpty())
+		})
+
+		It("Should create Operation to recreate Secret, since it is not present in Cluster.", func() {
+
+			defer dbq.CloseDatabase()
+
+			By("Set 'Created_on' field more than 30 Minutes.")
+
+			repositoryCredentials.Created_on = time.Now().Add(time.Duration(-(31 * time.Minute)))
+			err := dbq.UpdateRepositoryCredentials(ctx, &repositoryCredentials)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
Could this part here up to before the HaveLen check be commoned up with what is done in the previous test?